File: Remove Element/solve.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution
{
public:
    int removeElement(vector<int> &nums, int val)
    {
        int correctLength = 0;
        int initialLength = nums.size();
        for (int i = 0; i < initialLength; i++)
        {
            if (nums[i] != val)
            {
                nums[correctLength++] = nums[i];
            }
        }
        return correctLength;
    }
};

File: Remove Element/solve.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function (nums, val) {
    let numsLength = nums.length;
    let correctLength = 0;
    for (let i = 0; i < numsLength; i++) {
        if (nums[i] !== val) nums[correctLength++] = nums[i];
    }
    return correctLength;
};
